File: october/05-10/code.py
#** start of main.py **

def has_exoplanet(readings):

    minimun_average = average_readings(readings) * 80 / 100
    

    return found_exoplanet(readings, minimun_average)

def found_exoplanet(readings, average):
    found = False
    for element in readings:
        if (char_to_luminosity(element) <= average):
            found = True
            break

    return found

def average_readings(readings):
    count = 0
    for element in readings:
        count += char_to_luminosity(element)
         
    average = count / len(readings)

    return average

def char_to_luminosity(char):
    if 'A' <= char <= 'Z':
        return ord(char) - ord('A') + 10
    elif '0' <= char <= '9':
        return int(char)
    else:
        raise ValueError("Carácter inválido")



#** end of main.py **

File: october/05-10/code.js
//** start of script.js **

function hasExoplanet(readings) {
  let average =  minimunAverage(readings);
  //return readings;
  return foundExoplanet(readings,average);
}

function foundExoplanet(readings, average){
  let found = false;
  for(const element of readings){
    if(charToLuminosity(element) <= average){
      found = true;
      break;
    }
  }
  return found;
}

function minimunAverage(readings){
  let count = 0;
  let minimun;
  for(const element of readings){
    count += charToLuminosity(element);
  }
  minimun = (count / readings.length) * (80 / 100);
  console.log(minimun);
  return minimun;
}

function charToLuminosity(c) {
  if (c >= 'A' && c <= 'Z') {
    return c.charCodeAt(0) - 'A'.charCodeAt(0) + 10;
  } else if (c >= '0' && c <= '9') {
    return parseInt(c);
  } else {
    throw new Error("Carácter inválido");
  }
}



//** end of script.js **

